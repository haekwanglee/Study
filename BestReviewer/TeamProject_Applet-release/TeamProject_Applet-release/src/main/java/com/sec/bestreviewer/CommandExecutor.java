package com.sec.bestreviewer;

import com.sec.bestreviewer.command.Command;
import com.sec.bestreviewer.database.Table;
import com.sec.bestreviewer.store.Injection;

import java.util.List;

public class CommandExecutor {

    private final Table employeeDatabase;

    CommandExecutor() {
        this.employeeDatabase = Injection.provideEmployeeDatabase();
    }

    CommandExecutor(Table employeeDatabase) {
        this.employeeDatabase = employeeDatabase;
    }


    List<String> execute(Command command) {
        return command.execute(employeeDatabase);
    }
}