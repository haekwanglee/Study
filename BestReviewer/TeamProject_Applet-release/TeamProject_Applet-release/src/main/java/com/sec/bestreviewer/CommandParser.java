package com.sec.bestreviewer;


import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;


public class CommandParser {

    private final static int MIN_TOKENS_NUM = 5; // CMD + 3 of options + at least 1 params
    private final static String SPACE = " ";
    private final static String EMPTY = "";
    private final static String OPERATOR_AND = "-a";
    private final static String OPERATOR_OR = "-o";

    private List<String> tokenList;
    private List<SearchConditionPair> searchConditionPairs = new ArrayList<>();
    private List<String> conditionOptions, conditionParams, targetParams;
    private String type, optionMain, optionOperator;

    public TokenGroup parse(String line) throws IllegalArgumentException {
        getTokenListFromLine(line);
        initTokenize();
        tokenizeMain(tokenList);
        tokenizeParameter(tokenList);
        tokenizeMoreWhenHasOperator(tokenList);
        return new TokenGroup(type, optionMain, optionOperator, searchConditionPairs, targetParams);
    }

    private void initTokenize() {
        searchConditionPairs.clear();
    }

    private void getTokenListFromLine(String line) {
        tokenList = Arrays.asList(line.split(",", -1));
        if (tokenList.size() < MIN_TOKENS_NUM) {
            throw new IllegalArgumentException("wrong command format");
        }
    }

    private void tokenizeMain(List<String> tokenList) {
        type = getType(tokenList);
        optionMain = getOptionMain(tokenList);
        optionOperator = getOptionOperator(tokenList);
    }

    private void tokenizeParameter(List<String> tokenList) {
        int optionOperatorIndex = getOptionOperatorIndex(tokenList);
        conditionParams = getConditionParams(tokenList, type, optionOperatorIndex);
        conditionOptions = getConditionOptions(tokenList);
        targetParams = getTargetParams(tokenList, type, conditionParams);
        if (!type.equals(CommandFactory.CMD_ADD)) {
            searchConditionPairs.add(new SearchConditionPair(getValidList(conditionOptions), getValidList(conditionParams)));
        }
    }

    private void tokenizeMoreWhenHasOperator(List<String> tokenList) {
        int optionOperatorIndex = getOptionOperatorIndex(tokenList);
        if (optionOperatorIndex == tokenList.size()) {
            return;
        }
        conditionOptions = tokenList.subList(optionOperatorIndex + 1, optionOperatorIndex + 3);
        conditionParams = tokenList.subList(optionOperatorIndex + 3, optionOperatorIndex + 5);
        searchConditionPairs.add(new SearchConditionPair(getValidList(conditionOptions), getValidList(conditionParams)));
        if (optionOperatorIndex + 5 < tokenList.size()) {
            targetParams = tokenList.subList(optionOperatorIndex + 5, tokenList.size());
        }
    }

    private String getType(List<String> tokenList) {
        if (!isValidToken(tokenList.get(0))) {
            return "";
        }
        return tokenList.get(0);
    }

    private String getOptionMain(List<String> tokenList) {
        if (!isValidToken(tokenList.get(1))) {
            return "";
        }
        return tokenList.get(1);
    }

    private boolean isValidToken(String token) {
        if (EMPTY.equals(token) || SPACE.equals(token)) {
            return false;
        }
        return true;
    }

    private List<String> getValidList(List<String> list) {
        List<String> validList = new ArrayList<String>();
        list.forEach(i -> {
            if (isValidToken(i)) {
                validList.add(i);
            }
        });
        return validList;
    }

    private int getOptionOperatorIndex(List<String> tokenList) {
        int optionOperatorIndex = tokenList.size();
        for (String token : tokenList) {
            if (token.equals(OPERATOR_AND) || token.equals(OPERATOR_OR)) {
                optionOperatorIndex = tokenList.indexOf(token);
            }
        }
        return optionOperatorIndex;
    }

    private String getOptionOperator(List<String> tokenList) {
        String optionOperator = "";
        for (String token : tokenList) {
            if (token.equals(OPERATOR_AND) || token.equals(OPERATOR_OR)) {
                optionOperator = token;
            }
        }
        return optionOperator;
    }

    private List<String> getConditionOptions(List<String> tokenList) {
        return tokenList.subList(2, 4);
    }

    private List<String> getConditionParams(List<String> tokenList, String type, int endPosition) {
        List<String> params = tokenList.subList(4, endPosition);
        if (type.equals(CommandFactory.CMD_MOD)) {
            params = tokenList.subList(4, 6);
        }
        return params;
    }

    private List<String> getTargetParams(List<String> tokenList, String type, List<String> defaultParams) {
        List<String> targetParams = defaultParams;
        if (type.equals(CommandFactory.CMD_MOD)) {
            targetParams = tokenList.subList(6, 8);
        }
        return targetParams;
    }

}
