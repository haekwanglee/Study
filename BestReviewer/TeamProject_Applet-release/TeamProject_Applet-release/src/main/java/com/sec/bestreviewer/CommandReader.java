package com.sec.bestreviewer;

import java.io.BufferedReader;
import java.io.FileNotFoundException;
import java.io.FileReader;
import java.io.IOException;
import java.util.LinkedList;
import java.util.List;

public class CommandReader {

    public List<String> readFile(final String inputFileName) {

        final List<String> lineList = new LinkedList<>();

        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(inputFileName))) {
            String line = bufferedReader.readLine();

            while (line != null) {
                lineList.add(line);
                line = bufferedReader.readLine();
            }

        } catch (FileNotFoundException exception) {
            throw new IllegalArgumentException("Input file " + inputFileName + " is NOT exist");

        } catch (IOException exception) {
            exception.printStackTrace();
        }

        return lineList;
    }
}
