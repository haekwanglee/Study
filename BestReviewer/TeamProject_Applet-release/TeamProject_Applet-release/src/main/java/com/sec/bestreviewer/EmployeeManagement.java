package com.sec.bestreviewer;

import com.sec.bestreviewer.command.Command;

import java.util.List;

public class EmployeeManagement {

    private CommandReader commandReader;
    private final CommandParser commandParser;
    private final CommandExecutor commandExecutor;
    private final Printer printer;

    public EmployeeManagement() {
        commandReader = new CommandReader();
        commandParser = new CommandParser();
        commandExecutor = new CommandExecutor();
        printer = new Printer();
    }

    @Deprecated
    public void run(String[] args) {
        executeCommand(args[0]);
        printOutFile(args[1]);
    }

    public void printOutFile(String outputFileName) {
        printer.printOutputFile(outputFileName);
    }

    public void executeCommand(final String inputFileName) {

        final List<String> lineList = commandReader.readFile(inputFileName);
        lineList.forEach(this::executeEach);
    }

    public void executeEach(String line) {
        try {
            final TokenGroup tokenGroup = commandParser.parse(line);
            final Command command = CommandFactory.buildCommand(tokenGroup);
            final List<String> rowResult = commandExecutor.execute(command);

            printer.addCommandResult(tokenGroup, rowResult);
        } catch (IllegalArgumentException exception) {
            printer.onWrongMessage(line);
        }
    }

    public String peekCommandOutput() {
        return printer.peekStringOutput();
    }
}
