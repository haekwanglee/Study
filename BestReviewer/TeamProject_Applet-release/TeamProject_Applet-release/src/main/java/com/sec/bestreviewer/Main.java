package com.sec.bestreviewer;

public class Main {

    public static void main(String[] args) {

        if (!isValidArgument(args)) {
            printUsage();
            throw new IllegalArgumentException("wrong arguments count");
        }

        EmployeeManagement employeeManagement = new EmployeeManagement();
        employeeManagement.executeCommand(args[0]);
        employeeManagement.printOutFile(args[1]);
    }

    private static boolean isValidArgument(String[] args) {
        return args.length == 2;
    }

    private static void printUsage() {
        System.out.println("input / output 형태는 txt 파일이며, 아래 command 로 input file 을 read 하여, output file 을 생성한다.");
        System.out.println("Usage : EmployeeManagement input.txt output.txt");
    }
}
