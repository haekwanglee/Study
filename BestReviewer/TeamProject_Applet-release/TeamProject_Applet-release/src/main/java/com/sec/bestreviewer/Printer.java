package com.sec.bestreviewer;

import java.io.FileOutputStream;
import java.io.OutputStream;
import java.util.List;

public class Printer {

    private static final String COMMA = ",";
    private static final String NEW_LINE = "\n";
    private static final String NONE = "NONE";

    public static final String ERR_MSG_WRONG_COMMAND = "wrong command : ";
    public static final String FILE_READ_ERROR = "failed to read file : ";

    private final StringBuilder stdStringBuilder;

    public Printer() {
        this.stdStringBuilder = new StringBuilder();
    }

    public void onWrongMessage(final String line) {
        stdStringBuilder.append(ERR_MSG_WRONG_COMMAND);
        stdStringBuilder.append(line);
        stdStringBuilder.append(NEW_LINE);
    }

    public void addCommandResult(final TokenGroup tokenGroup, final List<String> dataList) {
        if (tokenGroup.getType().equals(CommandFactory.CMD_ADD)) {
            return;
        }

        if (dataList.isEmpty()) {
            appendNoneResult(tokenGroup.getType());
            return;
        }

        if (tokenGroup.hasOption(TokenGroup.Option.PRINT)) {
            final String result = printResultDetail(tokenGroup.getType(), dataList);
            stdStringBuilder.append(result);
        } else {
            final String result = printResult(tokenGroup.getType(), dataList);
            stdStringBuilder.append(result);
        }
    }

    void appendNoneResult(final String commandString) {
        stdStringBuilder.append(commandString).append(COMMA).append(NONE).append(NEW_LINE);
    }

    String printResult(final String commandString, final List<String> resultList) {
        return commandString + COMMA + resultList.size() + NEW_LINE;
    }

    String printResultDetail(final String commandString, final List<String> resultList) {
        final StringBuilder stringBuilder = new StringBuilder();

        resultList.stream()
                .limit(5)
                .forEachOrdered(resultLine -> {
                    stringBuilder.append(commandString);
                    stringBuilder.append(COMMA);
                    stringBuilder.append(resultLine);
                    stringBuilder.append(NEW_LINE);
                });

        return stringBuilder.toString();
    }

    public String peekStringOutput() {
        return stdStringBuilder.toString();
    }

    public void printOutputFile(final String outputFilename) {
        try (OutputStream output = new FileOutputStream(outputFilename)) {
            output.write(stdStringBuilder.toString().getBytes());
        } catch (Exception e) {
            appendFileReadFailure(e);
        }
    }

    private void appendFileReadFailure(Exception e) {
        stdStringBuilder.append(FILE_READ_ERROR)
                .append(e.getMessage())
                .append(NEW_LINE);
    }
}
