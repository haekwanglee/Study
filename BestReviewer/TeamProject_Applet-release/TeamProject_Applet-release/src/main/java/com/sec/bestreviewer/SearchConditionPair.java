package com.sec.bestreviewer;

import com.sec.bestreviewer.database.field.EmployeeSchema;

import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.function.BiFunction;

public class SearchConditionPair {

    private final List<String> options;
    private final List<String> params;

    SearchConditionPair(List<String> options, List<String> params) {
        this.options = options;
        this.params = params;
    }

    List<String> getOptions() {
        return options;
    }

    public List<String> getParams() {
        return params;
    }

    public boolean hasOptionSpecific(OptionSpecific option) {
        return option.hasOption(getOptions(), params.get(0));
    }

    public String getConditionTokenName() { //cl
        return params.get(0);
    }

    public String getConditionTokenValue() { //CL2
        return params.get(1);
    }

    public enum OptionSpecific {
        FIRST_OF_NAME((options, fieldParam) ->
                fieldParam.equals(EmployeeSchema.NAME.getName()) && options.contains("-f")),
        LAST_OF_NAME((options, fieldParam) ->
                fieldParam.equals(EmployeeSchema.NAME.getName()) && options.contains("-l")),

        MIDDLE_OF_PHONENUM((options, fieldParam) ->
                fieldParam.equals(EmployeeSchema.PHONE_NUMBER.getName()) && options.contains("-m")),
        LAST_OF_PHONENUM((options, fieldParam) ->
                fieldParam.equals(EmployeeSchema.PHONE_NUMBER.getName()) && options.contains("-l")),

        YEAR_OF_BIRTH((options, fieldParam) ->
                fieldParam.equals(EmployeeSchema.BIRTHDAY.getName()) && options.contains("-y")),
        MONTH_OF_BIRTH((options, fieldParam) ->
                fieldParam.equals(EmployeeSchema.BIRTHDAY.getName()) && options.contains("-m")),
        DAY_OF_BIRTH((options, fieldParam) ->
                fieldParam.equals(EmployeeSchema.BIRTHDAY.getName()) && options.contains("-d")),

        GREATER((options, fieldParam) -> options.contains("-g")),
        GREATER_EQUAL((options, fieldParam) -> options.contains("-ge")),
        SMALLER((options, fieldParam) -> options.contains("-s")),
        SMALLER_EQUAL((options, fieldParam) -> options.contains("-se"));

        private final BiFunction<List<String>, String, Boolean> checkOperator;

        OptionSpecific(BiFunction<List<String>, String, Boolean> checkOperator) {
            this.checkOperator = checkOperator;
        }

        public boolean hasOption(List<String> optionList, String fieldParam) {
            return checkOperator.apply(optionList, fieldParam);
        }
    }

    private static Map<OptionSpecific, String> operatorOptionToFieldName = new HashMap<>();

    static {
        operatorOptionToFieldName.put(SearchConditionPair.OptionSpecific.FIRST_OF_NAME,
                EmployeeSchema.NAME_FIRST.getName());
        operatorOptionToFieldName.put(SearchConditionPair.OptionSpecific.LAST_OF_NAME,
                EmployeeSchema.NAME_LAST.getName());
        operatorOptionToFieldName.put(SearchConditionPair.OptionSpecific.MIDDLE_OF_PHONENUM,
                EmployeeSchema.PHONE_NUMBER_MIDDLE.getName());
        operatorOptionToFieldName.put(SearchConditionPair.OptionSpecific.LAST_OF_PHONENUM,
                EmployeeSchema.PHONE_NUMBER_LAST.getName());
        operatorOptionToFieldName.put(SearchConditionPair.OptionSpecific.YEAR_OF_BIRTH,
                EmployeeSchema.BIRTH_DAY_YEAR.getName());
        operatorOptionToFieldName.put(SearchConditionPair.OptionSpecific.MONTH_OF_BIRTH,
                EmployeeSchema.BIRTH_DAY_MONTH.getName());
        operatorOptionToFieldName.put(SearchConditionPair.OptionSpecific.DAY_OF_BIRTH,
                EmployeeSchema.BIRTH_DAY_DAY.getName());
    }

    public String getConditionFieldName() {
        for (OptionSpecific optionKey : operatorOptionToFieldName.keySet()) {
            if (hasOptionSpecific(optionKey)) {
                return operatorOptionToFieldName.get(optionKey);
            }
        }
        return getConditionTokenName();
    }
}
