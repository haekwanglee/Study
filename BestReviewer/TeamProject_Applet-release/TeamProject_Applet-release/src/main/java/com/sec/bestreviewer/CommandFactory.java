package com.sec.bestreviewer;

import com.sec.bestreviewer.command.*;

import java.util.ArrayList;
import java.util.List;

public class CommandFactory {

    public static final String CMD_ADD = "ADD";
    public static final String CMD_DEL = "DEL";
    public static final String CMD_SCH = "SCH";
    public static final String CMD_MOD = "MOD";

    static Command buildCommand(TokenGroup tokenGroup)
            throws IllegalArgumentException {

        switch (tokenGroup.getType()) {
            case CMD_ADD:
                return new AddCommand(tokenGroup);
            case CMD_DEL:
                return new DeleteCommand(tokenGroup);
            case CMD_SCH:
                return new SearchCommand(tokenGroup);
            case CMD_MOD:
                return new ModCommand(tokenGroup);
        }
        throw new IllegalArgumentException("Wrong command");
    }
}