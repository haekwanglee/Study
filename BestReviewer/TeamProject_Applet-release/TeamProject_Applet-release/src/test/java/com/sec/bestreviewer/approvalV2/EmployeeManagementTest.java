package com.sec.bestreviewer.approvalV2;

import com.sec.bestreviewer.EmployeeManagement;
import org.approvaltests.Approvals;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import java.io.File;

/**
 * Test For added single options for phoneNumber, name, and birth
 */
public class EmployeeManagementTest {

    private static final String PATH = "./src/test/java/com/sec/bestreviewer/approvalV2/";

    @Test
    @DisplayName("성명의 이름(first name)이 E 보다 사전 순으로 뒤이거나 같은 record 검색")
    public void searchTest1() throws Exception {
        final String inputFileName = PATH + "EmployeeManagementTest.searchTest1.input.txt";
        final String outputFileName = PATH + "EmployeeManagementTest.searchTest1.output.txt";

        EmployeeManagement employeeManagement = new EmployeeManagement();

        File outputFile = new File(outputFileName);
        String[] args = {inputFileName, outputFileName};

        employeeManagement.run(args);

        Approvals.verify(outputFile);
    }

    @Test
    @DisplayName("성명의 이름(first name)이 E 보다 사전 순으로 큰 record 검색")
    public void testGreat() throws Exception {
        final String inputFileName = PATH + "EmployeeManagementTest.testGreat.input.txt";
        final String outputFileName = PATH + "EmployeeManagementTest.testGreat.output.txt";

        EmployeeManagement employeeManagement = new EmployeeManagement();

        File outputFile = new File(outputFileName);
        String[] args = {inputFileName, outputFileName};

        employeeManagement.run(args);

        Approvals.verify(outputFile);
    }

    @Test
    @DisplayName("경력개발단계가 CL3 보다 작은 record를 검색")
    public void searchTest2() throws Exception {
        final String inputFileName = PATH + "EmployeeManagementTest.searchTest2.input.txt";
        final String outputFileName = PATH + "EmployeeManagementTest.searchTest2.output.txt";

        EmployeeManagement employeeManagement = new EmployeeManagement();

        File outputFile = new File(outputFileName);
        String[] args = {inputFileName, outputFileName};

        employeeManagement.run(args);

        Approvals.verify(outputFile);
    }

    @Test
    @DisplayName("전화번호의 중간자리가 5555 보다 작거나 같은 record를 검색")
    public void searchTest3() throws Exception {
        final String inputFileName = PATH + "EmployeeManagementTest.searchTest3.input.txt";
        final String outputFileName = PATH + "EmployeeManagementTest.searchTest3.output.txt";

        EmployeeManagement employeeManagement = new EmployeeManagement();

        File outputFile = new File(outputFileName);
        String[] args = {inputFileName, outputFileName};

        employeeManagement.run(args);

        Approvals.verify(outputFile);
    }

    @Test
    @DisplayName("생년월일 중 연도가 2000보다 크거나 같은 record를 검색")
    public void searchTest4() throws Exception {
        final String inputFileName = PATH + "EmployeeManagementTest.searchTest4.input.txt";
        final String outputFileName = PATH + "EmployeeManagementTest.searchTest4.output.txt";

        EmployeeManagement employeeManagement = new EmployeeManagement();

        File outputFile = new File(outputFileName);
        String[] args = {inputFileName, outputFileName};

        employeeManagement.run(args);

        Approvals.verify(outputFile);
    }

    @Test
    @DisplayName("certi 가 PRO 보다 높거나 같은 record를 검색")
    public void searchTest5() throws Exception {
        final String inputFileName = PATH + "EmployeeManagementTest.searchTest5.input.txt";
        final String outputFileName = PATH + "EmployeeManagementTest.searchTest5.output.txt";

        EmployeeManagement employeeManagement = new EmployeeManagement();

        File outputFile = new File(outputFileName);
        String[] args = {inputFileName, outputFileName};

        employeeManagement.run(args);

        Approvals.verify(outputFile);
    }

    @Test
    @DisplayName("이름(First Name)이 F인 record를 삭제")
    public void nameOptionTest1() throws Exception {
        final String inputFileName = PATH + "EmployeeManagementTest.nameOptionTest1.input.txt";
        final String outputFileName = PATH + "EmployeeManagementTest.nameOptionTest1.output.txt";

        EmployeeManagement employeeManagement = new EmployeeManagement();

        File outputFile = new File(outputFileName);
        String[] args = {inputFileName, outputFileName};

        employeeManagement.run(args);

        Approvals.verify(outputFile);
    }

    @Test
    @DisplayName("성(Last Name)이 A인 전화번호를 010-0000-0000 으로 변경")
    public void nameOptionTest2() throws Exception {
        final String inputFileName = PATH + "EmployeeManagementTest.nameOptionTest2.input.txt";
        final String outputFileName = PATH + "EmployeeManagementTest.nameOptionTest2.output.txt";

        EmployeeManagement employeeManagement = new EmployeeManagement();

        File outputFile = new File(outputFileName);
        String[] args = {inputFileName, outputFileName};

        employeeManagement.run(args);

        Approvals.verify(outputFile);
    }

    @Test
    @DisplayName("전화 번호의 중간자리가 2222인 record를 삭제")
    public void phoneNumberTest1() throws Exception {
        final String inputFileName = PATH + "EmployeeManagementTest.phoneNumberTest1.input.txt";
        final String outputFileName = PATH + "EmployeeManagementTest.phoneNumberTest1.output.txt";

        EmployeeManagement employeeManagement = new EmployeeManagement();

        File outputFile = new File(outputFileName);
        String[] args = {inputFileName, outputFileName};

        employeeManagement.run(args);

        Approvals.verify(outputFile);
    }

    @Test
    @DisplayName("전화 번호의 뒷자리가 6666인 record의 certi를 PRO로 변경")
    public void phoneNumberTest2() throws Exception {
        final String inputFileName = PATH + "EmployeeManagementTest.phoneNumberTest2.input.txt";
        final String outputFileName = PATH + "EmployeeManagementTest.phoneNumberTest2.output.txt";

        EmployeeManagement employeeManagement = new EmployeeManagement();

        File outputFile = new File(outputFileName);
        String[] args = {inputFileName, outputFileName};

        employeeManagement.run(args);

        Approvals.verify(outputFile);
    }

    @Test
    @DisplayName("생년월일의 연도가 2001인 record의 성명을 Z Z로 변경")
    public void birthTest1() throws Exception {
        final String inputFileName = PATH + "EmployeeManagementTest.birthTest1.input.txt";
        final String outputFileName = PATH + "EmployeeManagementTest.birthTest1.output.txt";

        EmployeeManagement employeeManagement = new EmployeeManagement();

        File outputFile = new File(outputFileName);
        String[] args = {inputFileName, outputFileName};

        employeeManagement.run(args);

        Approvals.verify(outputFile);
    }

    @Test
    @DisplayName("생년월일의 월이 11인 record 의 경력개발단계를 CL4로 변경")
    public void birthTest2() throws Exception {
        final String inputFileName = PATH + "EmployeeManagementTest.birthTest2.input.txt";
        final String outputFileName = PATH + "EmployeeManagementTest.birthTest2.output.txt";

        EmployeeManagement employeeManagement = new EmployeeManagement();

        File outputFile = new File(outputFileName);
        String[] args = {inputFileName, outputFileName};

        employeeManagement.run(args);

        Approvals.verify(outputFile);
    }

    @Test
    @DisplayName("생년월일의 일이 11인 record 의 certi를 EX로 변경")
    public void birthTest3() throws Exception {
        final String inputFileName = PATH + "EmployeeManagementTest.birthTest3.input.txt";
        final String outputFileName = PATH + "EmployeeManagementTest.birthTest3.output.txt";

        EmployeeManagement employeeManagement = new EmployeeManagement();

        File outputFile = new File(outputFileName);
        String[] args = {inputFileName, outputFileName};

        employeeManagement.run(args);

        Approvals.verify(outputFile);
    }

    @Test
    @DisplayName("성명의 이름이 F이고, 경력개발단계가 CL3인 record를 삭제")
    public void multipleOptionDeleteTest1() throws Exception {
        final String inputFileName = PATH + "EmployeeManagementTest.multipleOptionDeleteTest1.input.txt";
        final String outputFileName = PATH + "EmployeeManagementTest.multipleOptionDeleteTest1.output.txt";

        EmployeeManagement employeeManagement = new EmployeeManagement();

        File outputFile = new File(outputFileName);
        String[] args = {inputFileName, outputFileName};

        employeeManagement.run(args);

        Approvals.verify(outputFile);
    }

    @Test
    @DisplayName(" 전화번호의 가운데가 3333이거나, 생년월일의 연도가 1900인 record를 삭제")
    public void multipleOptionDeleteTest2() throws Exception {
        final String inputFileName = PATH + "EmployeeManagementTest.multipleOptionDeleteTest2.input.txt";
        final String outputFileName = PATH + "EmployeeManagementTest.multipleOptionDeleteTest2.output.txt";

        EmployeeManagement employeeManagement = new EmployeeManagement();

        File outputFile = new File(outputFileName);
        String[] args = {inputFileName, outputFileName};

        employeeManagement.run(args);

        Approvals.verify(outputFile);
    }

    @Test
    @DisplayName("경력개발단계가 CL3, 생년월일의 월이 01인 record의 성명을 Z Z로 변경")
    public void multipleOptionModTest1() throws Exception {
        final String inputFileName = PATH + "EmployeeManagementTest.multipleOptionModTest1.input.txt";
        final String outputFileName = PATH + "EmployeeManagementTest.multipleOptionModTest1.output.txt";

        EmployeeManagement employeeManagement = new EmployeeManagement();

        File outputFile = new File(outputFileName);
        String[] args = {inputFileName, outputFileName};

        employeeManagement.run(args);

        Approvals.verify(outputFile);
    }

    @Test
    @DisplayName("생년월일의 일이 31이거나, certi가 PRO인 record의 생년월일을 20200202로 변경")
    public void multipleOptionModTest2() throws Exception {
        final String inputFileName = PATH + "EmployeeManagementTest.multipleOptionModTest2.input.txt";
        final String outputFileName = PATH + "EmployeeManagementTest.multipleOptionModTest2.output.txt";

        EmployeeManagement employeeManagement = new EmployeeManagement();

        File outputFile = new File(outputFileName);
        String[] args = {inputFileName, outputFileName};

        employeeManagement.run(args);

        Approvals.verify(outputFile);
    }

    @Test
    @DisplayName("성명의 이름이 F이고, 경력개발단계가 CL4인 record를 검색")
    public void multipleOptionSearchTest1() throws Exception {
        final String inputFileName = PATH + "EmployeeManagementTest.multipleOptionSearchTest1.input.txt";
        final String outputFileName = PATH + "EmployeeManagementTest.multipleOptionSearchTest1.output.txt";

        EmployeeManagement employeeManagement = new EmployeeManagement();

        File outputFile = new File(outputFileName);
        String[] args = {inputFileName, outputFileName};

        employeeManagement.run(args);

        Approvals.verify(outputFile);
    }

    @Test
    @DisplayName("전화번호의 가운데가 4444이거나, 생년월일의 연도가 2001인 record를 검색")
    public void multipleOptionSearchTest2() throws Exception {
        final String inputFileName = PATH + "EmployeeManagementTest.multipleOptionSearchTest2.input.txt";
        final String outputFileName = PATH + "EmployeeManagementTest.multipleOptionSearchTest2.output.txt";

        EmployeeManagement employeeManagement = new EmployeeManagement();

        File outputFile = new File(outputFileName);
        String[] args = {inputFileName, outputFileName};

        employeeManagement.run(args);

        Approvals.verify(outputFile);
    }

    @Test
    @DisplayName("전화번호의 가운데가 4444이거나, 생년월일의 연도가 2001보다 작은 record를 검색")
    public void multipleOptionSearchTest3() throws Exception {
        final String inputFileName = PATH + "EmployeeManagementTest.multipleOptionSearchTest3.input.txt";
        final String outputFileName = PATH + "EmployeeManagementTest.multipleOptionSearchTest3.output.txt";

        EmployeeManagement employeeManagement = new EmployeeManagement();

        File outputFile = new File(outputFileName);
        String[] args = {inputFileName, outputFileName};

        employeeManagement.run(args);

        Approvals.verify(outputFile);
    }

    @Test
    @DisplayName("And 조건에서 하나를 못 찾을 경우 정상처리 되는지 확인")
    public void multipleOptionSearchTest4() throws Exception {
        final String inputFileName = PATH + "EmployeeManagementTest.multipleOptionSearchTest4.input.txt";
        final String outputFileName = PATH + "EmployeeManagementTest.multipleOptionSearchTest4.output.txt";

        EmployeeManagement employeeManagement = new EmployeeManagement();

        File outputFile = new File(outputFileName);
        String[] args = {inputFileName, outputFileName};

        employeeManagement.run(args);

        Approvals.verify(outputFile);
    }

    @Test
    @DisplayName("And 조건에서 둘다 못 찾을 경우 정상처리 되는지 확인")
    public void multipleOptionSearchTest5() throws Exception {
        final String inputFileName = PATH + "EmployeeManagementTest.multipleOptionSearchTest5.input.txt";
        final String outputFileName = PATH + "EmployeeManagementTest.multipleOptionSearchTest5.output.txt";

        EmployeeManagement employeeManagement = new EmployeeManagement();

        File outputFile = new File(outputFileName);
        String[] args = {inputFileName, outputFileName};

        employeeManagement.run(args);

        Approvals.verify(outputFile);
    }

    @Test
    @DisplayName("OR 조건에서 하나를 못 찾을 경우 정상처리 되는지 확인")
    public void multipleOptionSearchTest6() throws Exception {
        final String inputFileName = PATH + "EmployeeManagementTest.multipleOptionSearchTest6.input.txt";
        final String outputFileName = PATH + "EmployeeManagementTest.multipleOptionSearchTest6.output.txt";

        EmployeeManagement employeeManagement = new EmployeeManagement();

        File outputFile = new File(outputFileName);
        String[] args = {inputFileName, outputFileName};

        employeeManagement.run(args);

        Approvals.verify(outputFile);
    }

    @Test
    @DisplayName("OR 조건에서 둘다 못 찾을 경우 정상처리 되는지 확인")
    public void multipleOptionSearchTest7() throws Exception {
        final String inputFileName = PATH + "EmployeeManagementTest.multipleOptionSearchTest7.input.txt";
        final String outputFileName = PATH + "EmployeeManagementTest.multipleOptionSearchTest7.output.txt";

        EmployeeManagement employeeManagement = new EmployeeManagement();

        File outputFile = new File(outputFileName);
        String[] args = {inputFileName, outputFileName};

        employeeManagement.run(args);

        Approvals.verify(outputFile);
    }
}
