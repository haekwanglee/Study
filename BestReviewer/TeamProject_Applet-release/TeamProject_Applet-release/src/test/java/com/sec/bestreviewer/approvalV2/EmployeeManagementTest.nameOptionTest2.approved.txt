MOD,3
SCH,95000001,A A,CL1,010-0000-0000,19001111,ADV
SCH,98000001,B A,CL1,010-0000-0000,19501111,ADV
SCH,99000001,C A,CL1,010-0000-0000,19921111,ADV
